src/stravalib/model.py

```diff
@@ -938,7 +938,7 @@ class Segment(
     map: Optional[Map] = None
     athlete_segment_stats: Optional[AthleteSegmentStats] = None
     athlete_pr_effort: Optional[AthletePrEffort] = None
-    activity_type: Optional[Literal["Ride", "Run"]] = None
+    activity_type: Optional[ActivityType] = None
```

Comment:
@jsamoocha i'm pretty sure it's ok for us to relax this - as it's pretty much the way things were previously. but i also remember another part of this module where i had to add a literal type for one of the methods. i think @enadeau helped me with that. so i just want to make sure this fix will actually work

Reply:
oh perhaps that was athlete type?

Reply:
This same fix was applied in #369. It led to typing errors (which we see now in CI). I have no idea ATM if we have to choose between two "evils" or if there's a (typing) workaround.

Reply:
ahhh ok thanks. so we should merge following @enadeau commits below.